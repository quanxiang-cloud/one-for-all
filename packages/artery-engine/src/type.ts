import type { MutableRefObject, CSSProperties  } from 'react';
import type { BehaviorSubject } from 'rxjs';
import type { Artery, Node } from '@one-for-all/artery';

export interface Block<T extends BaseBlocksCommunicationState> {
  render: (props: BlockItemProps<T>) => JSX.Element;
  style?: CSSProperties;
  hide?: boolean;
  id?: string;
}

export interface BlockProps<T extends BaseBlocksCommunicationState> extends Block<T> {
  onUpdateLayer: (params: Omit<UpdateLayer, 'layerId'> & { layerId?: string }) => void;
}

export interface Layer<T extends BaseBlocksCommunicationState> {
  blocks: Array<Block<T>>;
  style?: CSSProperties;
  id?: string;
  hide?: boolean;
}

export interface LayerProps<T extends BaseBlocksCommunicationState> extends Layer<T> {
  zIndex: number;
  updateLayer: (params: UpdateLayer) => void;
}

export type LayerTransfer<T extends BaseBlocksCommunicationState> = (layer: Layer<T>) => Layer<T>;

export interface BaseBlocksCommunicationState {
  [key: string]: any;
}

export type BlocksCommunicationState<T extends BaseBlocksCommunicationState> = BehaviorSubject<T>;
export interface BlockItemProps<T extends BaseBlocksCommunicationState> {
  onChange: (artery: Artery) => void;
  artery: Artery;
  sharedState: T;
  onSharedStateChange: (path: string, value: any) => void;
  commands?: CommandNameRunnerMap;
  commandsHasNext: boolean;
  commandsHasPrev: boolean;
  activeNode?: Node;
  setActiveNode: (node?: Node) => void;
  generateNodeId: (prefix?: string) => string;
  onUpdateLayer: (params: Omit<UpdateLayer, 'blockId' | 'layerId'> & { layerId?: string }) => void;
  onUpdateBlock: (params: Omit<UpdateLayer, 'layerId'> & { layerId?: string }) => void;
}

export interface Props<T extends BaseBlocksCommunicationState> {
  artery: Artery;
  layers: Array<Layer<T>>;
  blocksCommunicationStateInitialValue: T;
}

export interface EngineState<T extends BaseBlocksCommunicationState> {
  arteryStore$?: BehaviorSubject<Artery>;
  blocksCommunicationState$?: BlocksCommunicationState<T>;
  activeNode?: Node;
  useCommandState?: UseCommandState;
}

interface RedoUndo {
  undo?: () => void;
  redo?: () => void;
}

export interface Command {
  name: string;
  // eslint-disable-next-line
  execute: (...args: any[]) => RedoUndo;
  initer?: () => ((() => void) | undefined);
  keyboard?: string | string[];
}

// eslint-disable-next-line
export type CommandExecuteWrapper = (...args: any[]) => void;
export type CommandNameRunnerMap = Record<string, CommandExecuteWrapper>;

export interface CommandState {
  currentUndoRedoIndex: number;
  redoUndoList: RedoUndo[];
  commandList: Array<Command>;
  commandNameRunnerMap: CommandNameRunnerMap;
  destroyList: Array<((() => void) | undefined)>;
}

export interface UseCommandState {
  commandStateRef: MutableRefObject<CommandState>;
  registry: (command: Command) => void;
  commandNameRunnerMap: CommandNameRunnerMap;
}

export interface UpdateLayer {
  layerId: string;
  blockId?: string;
  name: string;
  value: unknown;
}
