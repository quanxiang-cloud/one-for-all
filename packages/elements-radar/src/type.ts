export interface Rect {
  x: number;
  y: number;
  height: number;
  width: number;
}

export interface ElementRect {
  raw: DOMRectReadOnly;
  relativeRect: Rect;
}

export type Report = Map< HTMLElement, ElementRect>;
