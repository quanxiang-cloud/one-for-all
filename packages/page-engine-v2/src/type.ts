export interface Block<T extends BaseBlocksCommunicationState> {
  render: (props: BlockItemProps<T>) => JSX.Element;
  gridColumnStart?: string;
  gridColumnEnd?: string;
  gridRowStart?: string;
  gridRowEnd?: string;
  visible?: boolean;
}

export interface BlockProps<T extends BaseBlocksCommunicationState> extends Block<T> {
  engineId: string;
  setLayer: (transfer: LayerTransfer<T>) => void;
}

export interface Layer<T extends BaseBlocksCommunicationState> {
  blocks: Array<Block<T>>;
  gridTemplateColumns: string;
  gridTemplateRows: string;
  blocksCommunicationStateInitialValue: T;
}

export interface LayerProps<T extends BaseBlocksCommunicationState> extends Layer<T> {
  zIndex: number;
  engineId: string;
  setLayer: (transfer: LayerTransfer<T>) => void;
}

export type LayerTransfer<T extends BaseBlocksCommunicationState> = (layer: Layer<T>) => Layer<T>;

export interface BaseBlocksCommunicationState {
  activeNodeID: string;
}

export type BlocksCommunicationState<T extends BaseBlocksCommunicationState> = import('rxjs').BehaviorSubject<T>;
export type Schema = import('@one-for-all/schema-spec').Schema;
export type SchemaNode = import('@one-for-all/schema-spec').SchemaNode;
export interface BlockItemProps<T extends BaseBlocksCommunicationState> {
  engineId: string;
  onChange: (schema: Schema) => void;
  schema: Schema;
  blocksCommunicationState$: BlocksCommunicationState<T>;
  setLayer: (transfer: LayerTransfer<T>) => void;
}

export interface Props<T extends BaseBlocksCommunicationState> {
  schema: Schema;
  layers: Array<Layer<T>>;
}

export interface EngineState<T extends BaseBlocksCommunicationState> {
  schemaStore$: import('rxjs').BehaviorSubject<Schema>;
  blocksCommunicationState$?: BlocksCommunicationState<T>;
}
