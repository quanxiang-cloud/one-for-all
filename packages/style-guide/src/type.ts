export type { StyleSheetPlain, RulePlain, CssNodePlain } from 'css-tree';

export type ClassNameSchema = {
  selector: string;
  desc?: string;
  pseudo?: ClassNameSchema[];
  children?: ClassNameSchema[];
}

export type StyleConfigInterface = {
  title: string;
  componentProps?: Record<string, unknown>[] | Record<string, unknown>;
  rules: ClassNameSchema[];
}

export type ComponentSpec = {
  key: string;
  specs: StyleConfigInterface[];
  title: string;
}
