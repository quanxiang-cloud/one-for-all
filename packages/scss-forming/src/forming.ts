import { AST, FormingRule } from './types';
import toAST from './to-ast';
import toSCSS from './to-scss';

type Input = string | AST;

async function forming(input: Input, formingRules: FormingRule[]): Promise<{ scss: string; ast: AST; }> {
  if (typeof input === 'string') {
    const ast = await formingSCSS(input, formingRules);
    const scss = await toSCSS(ast);
    return { ast, scss };
  }

  const ast = await formingAST(input, formingRules);
  const scss = await toSCSS(ast);
  return { ast, scss };
}

export default forming;
