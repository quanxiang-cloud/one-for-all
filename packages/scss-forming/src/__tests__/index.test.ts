jest.mock('../utils/to-formatted-scss');

import fs from 'fs';
import path from 'path';

import forming from '../index';
import type { FormingRule } from '../types';

const formingRules: FormingRule[] = [
  {
    selector: '.parent',
    nested: [
      { selector: '.children' }
    ]
  }
];

test('forming', async () => {
  const scssStr = fs.readFileSync(path.join(__dirname, 'fixtures/index.scss'), { encoding: 'utf-8'});
  const { scss } = await forming(scssStr, formingRules);

  expect(scss).toMatchSnapshot();
});
