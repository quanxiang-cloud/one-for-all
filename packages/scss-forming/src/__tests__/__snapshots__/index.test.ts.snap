// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forming 1`] = `
".parent{

  @include someMixin;
  color:red;

      .children{ font-size:20px; }
}
"
`;
